runners/spark/src/main/java/org/apache/beam/runners/spark/translation/TransformTranslator.java

```diff
@@ -130,17 +132,14 @@ public void evaluate(GroupByKey<K, V> transform, EvaluationContext context) {
             WindowedValue.FullWindowedValueCoder.of(coder.getValueCoder(), windowFn.windowCoder());
 
         // --- group by key only.
+        Long bundleSize =
+            context.getSerializableOptions().get().as(SparkPipelineOptions.class).getBundleSize();
+        Partitioner partitioner =
```

Comment:
It should be flipped to maintain old functionality:
```java
        Partitioner partitioner =
            (bundleSize > 0)
                ? null
                : new HashPartitioner(context.getSparkContext().defaultParallelism());
```

Reply:
I also think it will have basically the same effect if we always use null in the batch context. If we split a source on bundleSize and it has `n` partitions or split it on defaultParallelism and it has `defaultParallelism` partitions, either way going forward we want to maintain that many partitions which is what no `HashPartitioner` does. Only case where this is weird is if we try to split on defaultParallelism but the source doesn't support splitting that much causing `< defaultParallelism` partitions.